shared/AppInsightsCore/src/JavaScriptSDK/EnvUtils.ts

```diff
@@ -354,3 +357,140 @@ export function sendCustomEvent(evtName: string, cfg?: any, customDetails?: any)
     }
     return false;
 }
+
+/**
+ * Redacts user information from a URL
+ * @param url - The URL string to redact
+ * @returns The URL with user information redacted
+ */
+function redactUserInfo(url: string): string {
+    const schemeEndIndex = url.indexOf(":");
+    if (schemeEndIndex === -1) {
+        // not a valid url
+        return url;
+    }
+    const len = url.length;
+    if (len <= schemeEndIndex + 2
+        || url.charAt(schemeEndIndex + 1) !== "/"
+        || url.charAt(schemeEndIndex + 2) !== "/") {
+        return url;
+    }
+    
+    let index: number;
+    let atIndex = -1;
+    for (index = schemeEndIndex + 3; index < len; index++) {
+        const c = url.charAt(index);
+
+        if (c === "@") {
+            atIndex = index;
+        }
+
+        if (c === "/" || c === "?" || c === "#") {
+            break;
+        }
+    }
+    if (atIndex === -1 || atIndex === len - 1) {
+        return url;
+    }
+    return url.substring(0, schemeEndIndex + 3) + "REDACTED:REDACTED" + url.substring(atIndex);
+}
+
+/**
+ * Redacts sensitive query parameters from a URL
+ * @param url - The URL string to redact
+ * @returns The URL with sensitive query parameters redacted
+ */
+function redactQueryParameters(url: string, config?: IConfiguration): string {
```

Comment:
maybe we can pass sensitive_params string[] instead of the config object itself here?